ember-toucan-core/src/components/form/fields/checkbox-group.hbs

```diff
@@ -1,20 +1,41 @@
-<Form::Fieldset
-  @label={{@label}}
-  @hint={{@hint}}
-  @error={{@error}}
-  @isDisabled={{@isDisabled}}
-  aria-invalid={{if @error "true"}}
-  ...attributes
->
-  {{yield
-    (hash
-      CheckboxField=(component
-        (ensure-safe-component this.CheckboxFieldComponent)
-        name=@name
-        onChange=this.handleInput
-        isDisabled=@isDisabled
-        selectedValues=@value
-      )
-    )
-  }}
-</Form::Fieldset>
\ No newline at end of file
+<Form::Field as |field|>
+  <fieldset
```

Comment:
For my understanding: what was the reason to not use `<Form::Fieldset>` anymore?

Reply:
We need to be able to put `{{yield to="label"}}` and `{{yield to="hint}}` inside of say, the block for `field.Label` and the `Fieldset` component does not yield these subcomponents.

Reply:
I'm wondering if something similar to what I'm thinking for the form/core wrapper may work here.  I was hacking on this last night quite a bit and think I found something that'd work, although it's not necessarily pretty.  I'm curious if @simonihmig knows if there's a way to conditionally render named blocks when you have two-levels of components (e.g., `<Fieldset` exposes named blocks, but then another consuming component like `CheckboxGroup` wants to expose those underlying named blocks as well).

You can't conditionally render named-blocks inside of a block, **but** you _can_ conditionally render components.  So something really ugly like https://github.com/CrowdStrike/ember-toucan-core/commit/3ed2f97d7180d92ba5a6472dc7dca6946b379e7c#diff-68aa5e5601f7498e493199aca7a867ff41712f515756a155c3f7dcf1ccd45a10 may give us the best of both worlds?  Where folks can provide component arguments _or_ named blocks _or_ a combination of both.  That's what I was hacking on last night and it seems to work, although it's definitely not the best looking thing in the world.

For this case I think it'd look something like:

---

Update Fieldset to yield the new blocks like you have in this PR.

```hbs
<Form::Field as |field|>
  <fieldset
    aria-describedby="{{if @error field.errorId}} {{if @hint field.hintId}}"
    disabled={{@isDisabled}}
    ...attributes
  >
    {{! Use your new assert thing here instead, but just lazily posting the idea }}
    <legend
      class="type-md-tight text-body-and-labels block"
      data-label
    >{{@label}}{{yield to="label"}}</legend>

    {{! Use your new assert thing here instead, but just lazily posting the idea }}
    {{#if @hint}}
      <field.Hint id={{field.hintId}} data-hint>{{@hint}}{{yield to="hint"}}</field.Hint>
    {{/if}}

    <field.Control
      class="mt-1.5 flex flex-col space-y-2 rounded-sm p-1
        {{if @error 'shadow-error-outline'}}"
      data-control
    >
      {{yield}}
    </field.Control>

    {{#if @error}}
      <field.Error id={{field.errorId}} @error={{@error}} data-error />
    {{/if}}
  </fieldset>
</Form::Field>
```

---

And then in `CheckboxGroup`, use helpers to determine which mix of component arguments or named blocks are being used.  Map all properties over.

You can't do this:

```hbs
<Form::Fieldset
  @label={{@label}}
  @hint={{@hint}}
  @error={{@error}}
  @isDisabled={{@isDisabled}}
  aria-invalid={{if @error "true"}}
  ...attributes
>
  {{#if (has-block "label")}}
    {{yield to="label"}}
  {{/if}}

  {{#if (has-block "hint")}}
    {{yield to="hint"}}
  {{/if}}
  
  {{yield
    (hash
      CheckboxField=(component
        (ensure-safe-component this.CheckboxFieldComponent)
        name=@name
        onChange=this.handleInput
        isDisabled=@isDisabled
        selectedValues=@value
      )
    )
  }}
</Form::Fieldset>
```

But you could do something like this:

```hbs
{{#if (this.hasOnlyLabelBlock (has-block "label") (has-block "hint"))}}
  <Form::Fieldset
    @hint={{@hint}}
    @error={{@error}}
    @isDisabled={{@isDisabled}}
    aria-invalid={{if @error "true"}}
    ...attributes
  >
    <:label>{{yield to="label"}}</:label>
    {{yield
      (hash
        CheckboxField=(component
          (ensure-safe-component this.CheckboxFieldComponent)
          name=@name
          onChange=this.handleInput
          isDisabled=@isDisabled
          selectedValues=@value
        )
      )
    }}
  </Form::Fieldset>
{{else if (this.hasHintAndLabelBlocks (has-block "label") (has-block "hint"))}}
  <Form::Fieldset
    @hint={{@hint}}
    @error={{@error}}
    @isDisabled={{@isDisabled}}
    aria-invalid={{if @error "true"}}
    ...attributes
  >
    <:label>{{yield to="label"}}</:label>
    <:hint>{{yield to="hint"}}</:hint>
    {{yield
      (hash
        CheckboxField=(component
          (ensure-safe-component this.CheckboxFieldComponent)
          name=@name
          onChange=this.handleInput
          isDisabled=@isDisabled
          selectedValues=@value
        )
      )
    }}
  </Form::Fieldset>
{{else if (this.hasLabelArgAndHintBlock @label (has-block "hint"))}}
  <Form::Fieldset
    @label={{@label}}
    @hint={{@hint}}
    @error={{@error}}
    @isDisabled={{@isDisabled}}
    aria-invalid={{if @error "true"}}
    ...attributes
  >
    <:hint>{{yield to="hint"}}</:hint>
    {{yield
      (hash
        CheckboxField=(component
          (ensure-safe-component this.CheckboxFieldComponent)
          name=@name
          onChange=this.handleInput
          isDisabled=@isDisabled
          selectedValues=@value
        )
      )
    }}
  </Form::Fieldset>
{{else}}
  <Form::Fieldset
    @label={{@label}}
    @hint={{@hint}}
    @error={{@error}}
    @isDisabled={{@isDisabled}}
    aria-invalid={{if @error "true"}}
    ...attributes
  >
    {{yield
      (hash
        CheckboxField=(component
          (ensure-safe-component this.CheckboxFieldComponent)
          name=@name
          onChange=this.handleInput
          isDisabled=@isDisabled
          selectedValues=@value
        )
      )
    }}
  </Form::Fieldset>
{{/if}}
```

With your helpers here being:

```ts
  hasOnlyLabelBlock = (hasLabel: boolean, hasHint: boolean) =>
    hasLabel && !hasHint;
  hasHintAndLabelBlocks = (hasLabel: boolean, hasHint: boolean) =>
    hasLabel && hasHint;
  hasLabelArgAndHintBlock = (hasLabel: boolean, hasHint: boolean) =>
    hasLabel && hasHint;
```

Reply:
There's also `hasHintArgAndLabelBlock` and `hasHintArgAndLabelArg` and `hasOnlyLabelArg` I think.

Reply:
> ⚠️ You can't do this: ⚠️

What exactly can't we do?

Your example had

```hbs
{{#if (has-block "label")}}
  {{yield to="label"}}
{{/if}}
```

We can't do this?

Or is it about something like this, which is not possible? 👇

```hbs
{{#if (has-block "label")}}
  <:label>
    {{yield to="label"}}
  </:label>
{{/if}}
```

If that's the case, what is happening there? Does Ember refuse using any `<:some-block>` wrapped in `{{#if}}`?

Reply:
Yes named blocks like `:label` can't be wrapped in conditionals. (Sorry, yesterday I was struggling to remember the exact details, but this is why!)

RFC issue: https://github.com/emberjs/rfcs/issues/735